### PR TITLE
common/WorkQueue: Add a function in treadpool for draining a given work queue

### PR DESCRIPTION
**Pull request by jinweiyi**
The function threadpool::drain(wq) drains work queues. As the code comments say: If the parameter(wq) is not NULL, blocks until the given work queue does not have any items left to process. The threadpool , however,  blocks until all threads are idle, even if the given wq is empty. A new function -drain2(wq)- added in threadpool can drain a given wq. It implements the same action with drain(wq) when wq is NULL.

Signed-off-by: jinweiyi <jinweiyi@cmss.chinamobile.com>

### CODE REVIEW

**Comment by liewegas** [comment]
Is there a call site that needs this?  Also, maybe you can update the drain() comment to note that it also blocks until all workers are idle (even ones doing other wq work, if i understand you correctly)?

**Comment by jinweiyi** [comment]
So far, although(i am not sure) most call sites  (usually  before ending a threadpool) do not need this, yet some site where wqs process diff works may need this in some new feature later. For example, wq A shall do works for long time , while wq B does not. So I add a new drain2() to implement this.

**Comment by dachary** [comment]
jenkins test this please (eio now ignored in master)

**Comment by jinweiyi** [comment]
@liewegas , hi sage. I just update the drain() to support draining a given wq and no comment changed. This update,  of course, has no affect on any other call site where drain is called.

File: src/common/WorkQueue.cc [hunk 1]
```diff
@@ -124,6 +124,7 @@ void ThreadPool::worker(WorkThread *wt)
 	void *item = wq->_void_dequeue();
 	if (item) {
 	  processing++;
+          _wq_processing[wq]++;
```
Comment (amitkumar50) [inline src/common/WorkQueue.cc]: @jinweiyi (NIT) Please consider to keep PR title < 50 characters for future PRs.
https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#title-of-pull-requests-and-title-of-commits

**Comment by tchaikov** [comment]
@jinweiyi agreed with sage, because i am not sure if this feature is worthy of this extra complexity, if there is not identified caller site using it. please reopen this PR if you find one.

**Pull request closed without merging**